flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/connectors/hive/HiveTableSource.java

```diff
@@ -261,6 +279,93 @@ public DynamicTableSource copy() {
+                            jobConf,
+                            hiveVersion,
+                            tablePath,
+                            catalogTable.getPartitionKeys(),
+                            remainingPartitions);
+            BulkFormat<RowData, HiveSourceSplit> defaultBulkFormat =
+                    sourceBuilder.createDefaultBulkFormat();
+            List<HiveSourceSplit> inputSplits =
+                    HiveSourceFileEnumerator.createInputSplits(
+                            0, hivePartitionsToRead, threadNum, jobConf);
+            if (inputSplits.size() != 0) {
+                if (defaultBulkFormat instanceof FileBasedStatisticsReportableInputFormat) {
+                    return ((FileBasedStatisticsReportableInputFormat) defaultBulkFormat)
+                            .reportStatistics(
+                                    inputSplits.stream()
+                                            .map(FileSourceSplit::path)
+                                            .collect(Collectors.toList()),
+                                    catalogTable.getSchema().toRowDataType());
+                } else {
+                    return getMapRedInputFormatStatistics(
+                            inputSplits, catalogTable.getSchema().toRowDataType());
+                }
+            } else {
+                return TableStats.UNKNOWN;
+            }
+
+        } catch (Exception e) {
+            return TableStats.UNKNOWN;
+        }
+    }
+
+    private TableStats getMapRedInputFormatStatistics(
+            List<HiveSourceSplit> inputSplits, DataType producedDataType) {
+        if (inputSplits.isEmpty()) {
+            return TableStats.UNKNOWN;
+        }
+        // TODO now we assume that one hive external table has only one storage file format
+        HiveTablePartition hiveTablePartition = inputSplits.get(0).getHiveTablePartition();
+        StorageDescriptor sd = hiveTablePartition.getStorageDescriptor();
+        List<Path> files =
+                inputSplits.stream().map(FileSourceSplit::path).collect(Collectors.toList());
+        InputFormat mapredInputFormat;
+        try {
+            mapredInputFormat =
+                    (InputFormat)
+                            Class.forName(
+                                            sd.getInputFormat(),
+                                            true,
+                                            Thread.currentThread().getContextClassLoader())
+                                    .newInstance();
+            if (mapredInputFormat instanceof TextInputFormat) {
+                // only support csv format.
+                return CsvFormatStatisticsReportUtil.getTableStatistics(files);
+            } else if (mapredInputFormat instanceof OrcInputFormat) {
+                return OrcFormatStatisticsReportUtil.getTableStatistics(
+                        files, producedDataType, jobConf);
+            } else if (mapredInputFormat instanceof MapredParquetInputFormat) {
```

Comment:
What if `DeprecatedParquetInputFormat`?
I think it'll be better to check by `sd.getSerdeInfo.getSerializationLib`just like  `HiveInputFormat#useParquetVectorizedRead`

Reply:
> What if `DeprecatedParquetInputFormat`? I think it'll be better to check by `sd.getSerdeInfo.getSerializationLib`just like `HiveInputFormat#useParquetVectorizedRead`

Ok, Thanks, I try to use `sd.getSerdeInfo.getSerializationLib`. By doing this, we can also avoid to use reflection.